impl/src/metadata.rs

```diff
@@ -158,6 +285,117 @@ mod tests {
     };
 
     let mut fetcher = CargoMetadataFetcher::default();
-    assert!(fetcher.fetch_metadata(&files).is_err());
+    assert!(fetcher.fetch_metadata(&files, true).is_err());
+  }
+
+  #[test]
+  fn test_fetching_src() {
+    let dir = tempfile::TempDir::new().unwrap();
+    let lockfile_dir = tempfile::TempDir::new().unwrap();
+
+    let (path, _files) = fetch_crate_src(
```

Comment:
It is important to me that we make sure no unit tests are cloning git repositories. 

This will probably require you to create abstractions which can inject dependencies. Specifically, [cargo_clone::Cloner](https://docs.rs/cargo-clone-crate/0.1.3/cargo_clone/struct.Cloner.html) probably needs to be wrapped and stubbed for testing.

The larger problem here is that you've not really created any abstractions at all.